fedimintd/src/fedimintd.rs

```diff
@@ -160,66 +170,116 @@ impl Fedimintd {
+            .init()
+            .unwrap();
+
+        let bitcoind_rpc = BitcoinRpcConfig::from_env_vars()?;
+
         Ok(Self {
+            opts,
+            bitcoind_rpc,
             server_gens: ServerModuleInitRegistry::new(),
             server_gen_params: ServerModuleConfigGenParamsRegistry::default(),
             version_hash: version_hash.to_owned(),
         })
     }
 
-    /// Attach a server module to the Fedimintd instance
-    pub fn with_module<T>(mut self, gen: T) -> Self
+    /// Attach a server module kind to the Fedimintd instance
+    ///
+    /// This makes `fedimintd` support additional module types (aka. kinds)
+    pub fn with_module_kind<T>(mut self, gen: T) -> Self
     where
         T: ServerModuleInit + 'static + Send + Sync,
     {
         self.server_gens.attach(gen);
         self
     }
 
-    /// Attach additional parameters for a server module to Fedimintd
-    pub fn with_extra_module_inits_params<P>(
-        mut self,
-        id: ModuleInstanceId,
-        kind: ModuleKind,
-        params: P,
-    ) -> Self
+    /// Get the version hash this `fedimintd` will report for diagnostic
+    /// purposes
+    pub fn version_hash(&self) -> &str {
+        &self.version_hash
+    }
+
+    /// Attach additional module instance with parameters
+    ///
+    /// Note: The `kind` needs to be added with [`Self::with_module_kind`] if
+    /// it's not the default one.
+    pub fn with_module_instance<P>(mut self, kind: ModuleKind, params: P) -> Self
     where
         P: ModuleInitParams,
     {
         self.server_gen_params
-            .attach_config_gen_params(id, kind, params);
+            .attach_config_gen_params(kind, params);
         self
     }
 
     /// Attach default server modules to Fedimintd instance
     pub fn with_default_modules(self) -> Self {
+        let network = self.opts.network;
+
+        let bitcoind_rpc = self.bitcoind_rpc.clone();
+        let finality_delay = self.opts.finality_delay;
         let s = self
-            .with_module(LightningInit)
-            .with_module(MintInit)
-            .with_module(WalletInit);
+            .with_module_kind(LightningInit)
+            .with_module_instance(
+                LightningInit::kind(),
+                LightningGenParams {
+                    local: LightningGenParamsLocal {
+                        bitcoin_rpc: bitcoind_rpc.clone(),
+                    },
+                    consensus: LightningGenParamsConsensus { network },
+                },
+            )
+            .with_module_kind(MintInit)
+            .with_module_instance(
+                MintInit::kind(),
+                MintGenParams {
+                    local: Default::default(),
+                    consensus: MintGenParamsConsensus::new(
+                        2,
+                        fedimint_mint_server::common::config::FeeConsensus::default(),
+                    ),
+                },
+            )
```

Comment:
Longer term I think I'd rather leave it to the setup tool (admin UI, CLI, …) to determine which modules to instantiate "by default". These parameters are only defaults anyway afaik.